doc/src/syntax-and-macros.md

```diff
@@ -0,0 +1,78 @@
+# Slosh/Lisp syntax and macros
+
+The core unit of a Lisp is a form:
+```slosh
+42
+```
+```slosh
++
+```
+```slosh
+(list 1 2)
+```
+```slosh
+;; The quote symbol means return the literal form after the character
+;; This is identical to the form above.
+'(1 2)
+```
+are forms. All
+lisp code is structured as a sequence of forms inside forms separated by whitespace.
+
+Expressions are enclosed in parentheses and Lisp uses prefix notation to parse expressions.
+A parenthesized expression is a list of forms: `(list 1 2 3 4)` where the first
+form, the head, should generally be a function or a macro and dictates what the expression will do,
+and the subsequent children are arguments to the head.
+
+## Comments
+
+Single-line comments start with a ; character and continue till the end of the line.
+Multi-line comments start with a #| and end with a |#
```

Comment:
Also #; will consume and discard the next 'thing', usually an sexp- mostly for debugging.  Also could mention #%...%# for docstrings but those are special and not general comments.